### PR TITLE
Update class management functions

### PR DESCRIPTION
**Pull request by acalys**
- Added foreign key `teacherId` and `class_code` properties to `Class` model
- Created `class_code`'s randomize code auto-generation function 
- Updated base class and mixins for ClassesController

### CODE REVIEW

File: backend/api/models/Class.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 class Class(models.Model):
+    teacherId = models.ForeignKey(User, on_delete=models.CASCADE)
     name = models.CharField(max_length=100)
     sections = models.CharField(max_length=100)
-    schedule = models.CharField(max_length=100)
\ No newline at end of file
+    schedule = models.CharField(max_length=100)
+    date_created = models.DateTimeField(auto_now=True)
+    date_updated = models.DateTimeField(auto_now=True)
+    class_code = models.CharField(max_length=100) # class code is generated when class is created
```
Comment (omega13-engr) [inline backend/api/models/Class.py]: Make this unique

File: backend/api/serializers/ClassSerializer.py [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
             'name': 'Class Name',
+            'teacherId': 'Teacher',
             'sections': 'Number of Sections',
             'schedule': 'Schedule',
-        }
\ No newline at end of file
+            'class_code': 'Class Code'
+        }
+
+    def create(self, validated_data):
```
Comment (omega13-engr) [inline backend/api/serializers/ClassSerializer.py:21]: should checks if class_code doesn't exist in the database

**Review by slaee — CHANGES_REQUESTED** [review]

File: backend/api/serializers/ClassSerializer.py (hunk 2 above)
Comment (acalys) [reply backend/api/serializers/ClassSerializer.py:21]: All requested changes are made. You may review them now.

File: backend/api/models/Class.py [hunk 3]
```diff
@@ -6,8 +6,9 @@ class Class(models.Model):
     name = models.CharField(max_length=100)
     sections = models.CharField(max_length=100)
     schedule = models.CharField(max_length=100)
-    date_created = models.DateTimeField(auto_now=True)
-    date_updated = models.DateTimeField(auto_now=True)
-    class_code = models.CharField(max_length=100) # class code is generated when class is created
+    date_created = models.DateTimeField(auto_now_add=True, unique=True)
```
Comment (omega13-engr) [inline backend/api/models/Class.py]: sayop ni alys hahahah ang class_code ra dapat unique eh revert ni

**Review by slaee — CHANGES_REQUESTED** [review]

**Review by slaee — APPROVED** [review]

**Review by slaee — APPROVED** [review]

File: backend/api/models/Class.py (hunk 3 above)
Comment (acalys) [reply backend/api/models/Class.py]: oki oki hahahahaha

**Pull request merged**